### PR TITLE
DEV-1612: inspect accepts reference as str | list[str] (batched single-kind lookup)

### PR DESCRIPTION
**Pull request by ZmeiGorynych**
## What

Widen the `inspect` point-lookup (`InspectService.inspect` + MCP / REST / CLI / `SlayerClient` surfaces) to accept **either a single id or a homogeneous-kind list of ids**. Pure additive widening — existing single-id callers are unaffected.

Spun off from DEV-1591 (bird-agents compact-search token fix): the prompts are migrating their targeted-detail reads onto `inspect`, but their reads are often multi-entity. Batching turns a loop of N single-id calls back into one round-trip.

## Behaviour

- **Single `str`** keeps its current behaviour and output **byte-for-byte**.
- **List** → one rendered block per id, in **input order**, each echoing its resolved canonical id:
  - markdown: a `## <canonical>` header per block, blocks joined by `\n\n---\n\n` (error blocks header = the input ref).
  - `format="json"`: a bare JSON array; success elements keep the per-kind `{canonical_id, entity_type, description, [text], warnings}` shape, failures are `{"reference": <input>, "error": <message>}` objects.
- **Framing is input-type-driven, not length-driven** — a one-element list is still batch-framed.
- **Per-id resolution errors are isolated** — one bad id does not sink the batch.
- **Global arg errors still raise** (bad `entity_type` / `format`, negative `descriptions_max_chars`, and an **empty list**), validated once before the loop.
- No dedup; duplicate ids → repeated blocks.

Internally a per-id `_inspect_one(...) -> _OneResult(canonical_id, is_error, serialized)` feeds both paths; the str path returns `.serialized` (byte-for-byte) and the batch path frames success vs error from the explicit `is_error` flag (never inferred).

## Surfaces

- MCP `inspect` tool signature + docstring.
- REST `InspectRequest.reference: str | list[str]` (empty list → 400, non-string member → 422); `{"result": <string>}` envelope unchanged (a list yields a JSON-array string).
- CLI positional `reference` → `nargs="+"` with a `len==1 → bare str` adapter.
- `SlayerClient.inspect` / `inspect_sync` + HTTP body builder.

## Tests

New `tests/test_inspect_batch.py` (core) + batch classes in `tests/test_inspect_surfaces.py`: single-id back-compat (incl. byte-for-byte + JSON-error-stays-plain-string), ordered valid blocks, one-bad-id isolation, kind mismatch as per-id error, all-error list, duplicates, compact True/False on a list, one-element-list framing, empty-list raise, global-arg raises, element-shape parity with single-id, datasource-separator robustness, per-element warnings scoping, `descriptions_max_chars` across a list, and all four surfaces (MCP schema/docstring, REST array body + 400/422, CLI multi vs single positional + nargs parse, client sync/async local/remote).

Full non-integration suite: 6156 passed, 0 failed. `ruff` clean.

Plan reviewed by Codex (approach + edge cases) and the test suite verified against the plan by Codex before implementation.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added batch inspection support for `inspect` using multiple same-kind references across the CLI, API, MCP tool, and Python client.
  * Batch outputs preserve input order with one markdown block per reference (and JSON-array responses for list input), including canonical-id echoing on success.

* **Bug Fixes**
  * Per-reference resolution failures are isolated (one bad id won’t break the whole batch).
  * Empty batches now error (HTTP 400); non-string list items error (HTTP 422); single-element lists use batch framing vs single strings.

* **Documentation**
  * Updated all relevant `inspect` interface/API/client docs to reflect list semantics and formatting.

* **Tests**
  * Added comprehensive batch/list coverage across surfaces.

### CODE REVIEW

File: tests/test_inspect_surfaces.py [hunk 1]
```diff
@@ -307,3 +311,291 @@ def _fake_request_sync(*, method, path, json=None, params=None):
...
+
+        monkeypatch.setattr(client, "_request", _fake_request)
+        out = await client.inspect(
+            reference=["mydb.orders.amount", "mydb.orders.customer_id"],
+            entity_type="column", compact=False,
+        )
+        assert out == "REMOTE BATCH ASYNC"
+        assert captured["json"]["reference"] == [
+            "mydb.orders.amount", "mydb.orders.customer_id",
+        ]
```
[the remark this answers is not in the document]
Comment (ZmeiGorynych) [reply tests/test_inspect_surfaces.py:610]: @coderabbitai These surface suites are not integration tests: they run entirely against an in-memory `YAMLStorage` with `sqlite:///:memory:` and require no external database, so the `@pytest.mark.integration` guideline (which is for tests needing a live DB, all of which live under `tests/integration/`) does not apply. They are also consistent with the pre-existing surface suites in this same file (`TestMcpSurface`, `TestRestSurface`, `TestCliSurface`, `TestSlayerClientSurface`), none of which are marked integration. Marking only the new classes would make the file inconsistent. Leaving as-is.
